src/storage/ArrayStorage.java

```diff
@@ -0,0 +1,93 @@
+package storage;
+
+import model.Resume;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+public class ArrayStorage {
+    private static final Logger log = Logger.getLogger("ArrayStorage Logger");
+
+    private static final int MAX_STORAGE_SIZE = 100000;
+    private final Resume[] storage = new Resume[MAX_STORAGE_SIZE];
+    private int size;
+
+    public void clear() {
+        Arrays.fill(storage, null);
+        size = 0;
+    }
+
+    public void save(Resume r) {
+        Objects.requireNonNull(r);
+        if (size == MAX_STORAGE_SIZE) {
+            log.warning("Storage max limit is reached. Please, clear storage to add new elements.");
+            return;
+        }
+        if (isPresent(r.getUuid())) {
+            log.warning(String.format("Resume(uuid=%s) is already present in storage. It may be updated only.%n", r.getUuid()));
+        } else {
+            storage[size] = r;
+            size++;
+        }
+    }
+
+    public void update(Resume r) {
+        Objects.requireNonNull(r);
+        if (isPresent(r.getUuid())) {
+            storage[getResumeIndexById(r.getUuid())] = r;
+        } else {
+            log.warning(String.format("Resume(uuid=%s) is not present in storage. Please, add it.%n", r.getUuid()));
+        }
+    }
+
+    public void delete(String uuid) {
```

Comment:
Same, you iterate twice to find a resume, it's too expensive

Reply:
Fixed